src/parser_json.c

```diff
@@ -452,6 +452,29 @@ json_get_value(struct lyd_node_leaf_list *leaf, struct lyd_node **first_sibling,
         if (data[len++] != '[') {
             LOGVAL(ctx, LYE_XML_INVAL, LY_VLOG_LYD, leaf, "JSON data (expected begin-array)");
             return 0;
+        } else {
+            unsigned int len2 = len;
+            // Strip whitespaces
+            for (r = len2 + 1; isspace(data[r]); ++r);
+            // Check if an empty list [] was provided
+            if (data[r] == ']') {
+                leaf->value_str = lydict_insert(ctx, "", 0);
+                len2 = r + 1;
+                len2 += skip_ws(&data[len2]);
+                return len2;
```

Comment:
deduplication: This code is duplicated again just below.  I suggest you combine the two, and simply set a flag "is_empty_list" in here, plus in the next branch.  Then test the flag to determine whether to run this code.